pkg/util/mock/context.go

```diff
@@ -273,14 +284,41 @@ func (c *Context) NewTxn(context.Context) error {
 	return nil
 }
 
-// NewStaleTxnWithStartTS implements the sessionctx.Context interface.
-func (c *Context) NewStaleTxnWithStartTS(ctx context.Context, _ uint64) error {
-	return c.NewTxn(ctx)
+// fakeTxn is used to let some tests pass in the context without an available kv.Storage. Once usages to access
+// transactions without a kv.Storage are removed, this type should also be removed.
+// New code should never use this.
+type fakeTxn struct {
```

Comment:
As [the fake too-large StartTS function](https://github.com/pingcap/tidb/pull/58054/files#diff-4216b91ef15e2ea8a93fb3d322a8fb0bc66e246fd55f6551a6aa4d96262203f3L140) looks not yet implemented in this version, did you try if it works without my changes in `mock/context.go`?

Reply:
No I didn't. Are you suggesting we should discard all changes in mock/context.go?